### ISSUE TITLE
Emmet wrapping with multiple selections in single line is extremely buggy and unreliable

### ISSUE THREAD

**Issue by iansan5653** [Microsoft/vscode#49138]
- VSCode Version: 1.22.2
- OS Version: Windows 10 Home v1709

Steps to Reproduce:

1. Select multiple words/phrases/tags on the same or different lines in an HTML file.
2. Invoke Emmet > Wrap Individual Lines With Abbreviation
3. Type something, then backspace, then type something else, then backspace. The code is completely ruined unless you undo.

Does this issue occur when all extensions are disabled?: Yes

**Comment by iansan5653** [comment]
Here's an example:

1. Start with `The quick brown fox jumps over the lazy dog.`
2. Select `quick` `fox` and `the lazy`. (By holding down <kbd>alt</kbd>).
3. Start Emmet Wrap.
4. Type `b`, <kbd>backspace</kbd>, `a`, <kbd>backspace</kbd>, `p`, <kbd>backspace</kbd>, <kbd>esc</kbd>.
5. Expected result is `The quick brown fox jumps over the lazy dog.`, but actual result is:
```
The quickef="|"><p>foxck</a>foxfox<p>the lazy="|">fox</a>e lazyb>the <a href="|">the lazy</a> dog.
```

**Comment by ramya-rao-a** [comment]
Thanks for reporting!
I've noticed that the issue is restricted to when the multiple selections are in a single line.
Can you confirm?

**Comment by ramya-rao-a** [comment]
@iansan5653 I have pushed a change to disable the live preview feature when there are multiple selections in the same line. This should unblock you from using the wrap feature.

@gushuro  The logic that calculates the preview range is not quite right when there are multiple selections in the same line and is causing this issue.

**Comment by ramya-rao-a** [comment]
@iansan5653 Thanks for confirming. The latest [Insiders](https://code.visualstudio.com/insiders) has the live preview of the wrap feature disabled for such cases. Give it a try.

**Comment by jeanp413** [comment]
@ramya-rao-a I'll take this up

**Comment by ramya-rao-a** [comment]
Fixed by @jeanp413 with #49501
